Lib/test/test_logging.py

```diff
@@ -683,7 +682,9 @@ def emit(self, record):
                     self.sub_handler.release()
 
         self.assertEqual(len(logging._handlers), 0)
-        refed_h = _OurHandler()
+        stream = open('/dev/null', 'wt')
+        self.addCleanup(stream.close)
```

Comment:
You can use self.addCleanup(refed_h.stream.close), the .stream attribute is public. No need to modify _OurHandler.

Reply:
Thanks Victor. I think it should have been `self.addCleanup(refed_h.sub_handlerstream.close)` since it's the sub_handler's stream that need to be closed. I have reverted my changes and added your suggestion.